python/lsst/sims/featureScheduler/basis_functions/feasibility_funcs.py

```diff
@@ -0,0 +1,172 @@
+import numpy as np
+from lsst.sims.featureScheduler import features
+import matplotlib.pylab as plt
+from lsst.sims.featureScheduler.basis_functions import Base_basis_function
+
+
+__all__ = ['Filter_loaded_basis_function', 'Time_to_twilight_basis_function',
+           'Not_twilight_basis_function', 'Force_delay_basis_function',
+           'Hour_Angle_limit_basis_function', 'Moon_down_basis_function',
+           'Fraction_of_obs_basis_function', 'Clouded_out_basis_function']
+
+
+class Filter_loaded_basis_function(Base_basis_function):
+    """Check that the filter(s) needed are loaded
+
+    Parameters
+    ----------
+    filternames : str or list of str
+        The filternames that need to be mounted to execute.
+    """
+    def __init__(self, filternames='r'):
+        super(Filter_loaded_basis_function, self).__init__()
+        if type(filternames) is not list:
+            filternames = [filternames]
+        self.filternames = filternames
+
+    def check_feasibility(self, conditions):
+
+        for filtername in self.filternames:
+            result = filtername in conditions.mounted_filters
+            if result is False:
+                return result
+        return result
+
+
+class Time_to_twilight_basis_function(Base_basis_function):
+    """Make sure there is enough time before twilight. Useful
+    if you want to check before starting a long sequence of observations.
+
+    Parameters
+    ----------
+    time_needed : float (30.)
+        The time needed to run a survey (mintues).
+    """
+    def __init__(self, time_needed=30.):
+        super(Time_to_twilight_basis_function, self).__init__()
+        self.time_needed = time_needed/60./24.  # To days
+
+    def check_feasibility(self, conditions):
+        available_time = conditions.sun_n18_rising - conditions.mjd
```

Comment:
Does this mean this is only until 18 degree twilight? Are there other twilight conditions in the conditions (so you could potentially pick which one you wanted to use, here)?